sklearn/metrics/_classification.py

```diff
@@ -670,7 +698,17 @@ class labels [2]_.
     .. [3] `Wikipedia entry for the Cohen's kappa
             <https://en.wikipedia.org/wiki/Cohen%27s_kappa>`_.
     """
-    confusion = confusion_matrix(y1, y2, labels=labels, sample_weight=sample_weight)
+    y_type, y1, y2 = _check_targets(y1, y2)
```

Comment:
Since `_check_targets` call `unique`, this adds more overhead to the computation.

REF: https://github.com/scikit-learn/scikit-learn/pull/26820

Reply:
I don't know how to go around that. Somehow having private function to call could help and passing the unique label could help but I would not block this PR for it.